CIP-0054/README.md

```diff
@@ -0,0 +1,227 @@
+
+In tandem with the additional metadata, we also define a standard for the Javascript API which is provided to the NFT within the sandbox.
+
+## The Metadata
+
+Minting metadata for Smart NFTs – based on the existing [CIP-0025](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0025) standard:
+
+```
+{
+	"721": {
+		"<policy_id>": {
+			"<asset_name>": {
+				"name": <string>,
+
+				"image": <uri | array>,
+				"mediaType": "image/<mime_sub_type>",
+
+				"description": <string | array>,
+
+				"files": [{
+					"id": <string>
+					"name": <string>,
+					"mediaType": <mime_type>,
+					"src": <uri | array>,
+					<other_properties>
+				}],
+
+				"uses": {
+					"transactions": <string | array>,
+					"tokens": <string | array>,
+					"renderer": <string>	  
+				}
+			}
+		},
+		"version": "1.0"
+	}
+}
+```
+
+Here we have added the “uses” key – any future additions to the Smart NFT API can be implemented by adding additional keys here. 
+We've also added an additional field to the files array - this is to specify a unique identifier to enable the files to be referenced by the Javascript API below. 
+
+### The transactions key
+
+To enable evolving NFTs where the NFT monitors a transaction history and changes in response to new transaction metadata, we define a sub-key called “transactions”, which can contain either a string or an array, specifying the tokens or addresses the NFT wishes to receive the transaction history for. These transaction histories will be provided to the NFT via the Javascript API detailed below.
+
+We also define a special keyword “own” which can be used to monitor the NFT’s own transaction history. So if we wish to create an “evolvable” NFT that can respond to additions to its own transaction history, the “uses” key within the metadata would look like:
+
+```
+	"uses": {
+		"transactions": "own"
+	}
+```
+
+If you wanted to create an evolving NFT which monitors its own transaction history, as well as that of an external smart contract address, the metadata would look like this:
+
+```
+ 	"uses": {
+		"transactions": [
+			"own",
+			"addr1wywukn5q6lxsa5uymffh2esuk8s8fel7a0tna63rdntgrysv0f3ms"
+		]			
+	}
+```
+Finally, we also provide the option to receive the transaction history for a specific token other than the NFT itself (generally this is intended to enable import of Oracle data or control data from an external source – although monitoring an address transaction history could also be used for that). 
+
+When specifying an external token to monitor, you should do so via the token’s fingerprint as in this example:
+
+```
+ 	"uses": {
+		"transactions": [
+			"own",			
+			"asset1frc5wn889lcmj5y943mcn7tl8psk98mc480v3j"
+		]
+	}
+```
```

Comment:
Both examples with the asset id and the address are heavily user-facing / JSON-specific. In principle, this metadata would be stored on-chain and I would advise against storing these things are text strings. Both addresses and asset ids have well-defined on-chain binary representation, use them! 

Also, regarding asset fingerprints, these are not meant to be global unique identifiers. They are fine in the context of a single wallet but they should not be used as id in a global setup. Rather, use the asset id (i.e. policy id + asset name). Incidentally, it would be _okay_ to however display the asset fingerprint to the user, so long as it is stored as an asset id.